### PR TITLE
Use release versioning

### PR DESCRIPTION
**Pull request by alexgg**

### CODE REVIEW

File: automation/entry_scripts/balena-deploy-block.sh [hunk 1]
```diff
@@ -23,28 +25,34 @@ if [ "$ESR" = "true" ]; then
...
+_releaseID=$(balena_lib_release "${BALENAOS_ACCOUNT}/$APPNAME" "${FINAL}" "/deploy" "${API_ENV}" "$_local_image")
+if [ -z "${_releaseID}" ]; then
+	echo "[INFO] Failed to deploy to ${BALENAOS_ACCOUNT}/$APPNAME"
+	exit 1
+fi
 
 # Legacy hostapp tagging
-release_version="${RELEASE_VERSION}"
-if [ -n "${VARIANT}" ]; then
+if [ "${FINAL}" = "yes" ] && [ -n "${VARIANT}" ]; then
```
Comment (floion) [inline automation/entry_scripts/balena-deploy-block.sh:37]: So all builds will get deployed as draft releases (even PRs ?) unless the build has the `deploy` jenkins build param set to `yes` in which case the release is deployed as final?
And the deployment as final should only happen at this time when the `BOARDNAME` jenkins job has been called by `yocto-deploy-board` job.

File: automation/include/balena-api.inc [hunk 2]
```diff
@@ -352,35 +352,97 @@ balena_api_is_dt_private() {
...
+	if balena_api_fcn_exists balena_lib_token; then
+		_token=${_token:-$(balena_lib_token)}
+	fi
+	if [ -n "${_token}" ]; then
+		_auth="-H \"Authorization: Bearer ${_token}\""
+	fi
+	_semver=${_release_version%+*}
+	_revision=${_release_version#*+}
+	__dlog "[${_appName}] Fetching release version  ${_semver}+${_revision}"
+	_json=$(${CURL} -XGET -H "Content-type: application/json" ${_auth:-} "https://api.${_apiEnv}/${TRANSLATION}/release?\$filter=(belongs_to__application%20eq%20${_appID})%20and%20(semver%20eq%20%27${_semver}%27)%20and%20(revision%20eq%20%27${_revisioin}%27)")
```
[the remark this answers is not in the document]
Comment (alexgg) [reply automation/include/balena-api.inc]: thanks @floion, it's fixed now. I hadn't yet tested this branch fully.

**Comment by floion** [comment]
@balena-ci rebase

File: automation/entry_scripts/balena-deploy-block.sh (hunk 1 above)
Comment (alexgg) [reply automation/entry_scripts/balena-deploy-block.sh]: @floion in current releases a hostapp is only deployed when the job is marked as `deploy`.

With the current state of this PR, this does not change. Hostapps are only deployed if the job is marked as `deploy. And as you well say, all deployments will be draft releases. They will need to be flagged as final externally, either manually or by Leviathan.

At the moment this has little implications as balenaCloud does not yet use the release version but the hostapp labels, and these are created for `deploy` jobs.

**Comment by alexgg** [comment]
@balena-ci I self-certify!

**Pull request merged**